tests/main/test_openai_wrapper.py

```diff
@@ -113,17 +102,303 @@ def test_predict_call(self):
+        history = self.HISTORY
+        answer = self.ANSWER
+        test_data = pd.DataFrame(data=[[params['context'], questions, answer, sys_prompt, history]],
+                                 columns=['context', 'questions', 'answer', 'sys_prompt', 'history'])
+        expected_content = self.EXPECTED_ANSWER
+        mock_result = self.create_mock_result(expected_content, params['is_openai_1_0'])
+        self.assert_result(params, mock_result, test_data)
+        self.assert_called_with(params, mock_result, test_data,
+                                sys_prompt=sys_prompt, history=history,
+                                questions=questions)
+
+    @pytest.mark.skipif(sys.version_info.minor <= 6,
+                        reason='Openai not supported for older versions')
+    def test_predict_call_df(self):
+        params = self.get_model_params()
+        questions = self.QUESTION
+        sys_prompt = self.SYS_PROMPT
+        history = self.HISTORY
+        expected_content = self.EXPECTED_ANSWER
+        mock_result = self.create_mock_result(expected_content, params['is_openai_1_0'])
+
+        # Only prompt
+        test_data = pd.DataFrame(data=[[params['context'], questions]],
+                                 columns=['context', 'questions'])
+        self.assert_called_with(params, mock_result, test_data,
+                                questions=questions)
+
+        # Prompt and sys_prompt
+        test_data = pd.DataFrame(data=[[params['context'], questions, sys_prompt]],
+                                 columns=['context', 'questions', 'sys_prompt'])
+        self.assert_called_with(params, mock_result, test_data,
+                                sys_prompt=sys_prompt, questions=questions)
+
+        # Prompt and history
+        test_data = pd.DataFrame(data=[[params['context'], questions, history]],
+                                 columns=['context', 'questions', 'history'])
+        self.assert_called_with(params, mock_result, test_data,
+                                history=history, questions=questions)
+
+        # Prompt, history, and sys_prompt
+        test_data = pd.DataFrame(data=[[params['context'], questions, history, sys_prompt]],
+                                 columns=['context', 'questions', 'history', 'sys_prompt'])
+        self.assert_called_with(params, mock_result, test_data,
+                                sys_prompt=sys_prompt, history=history,
+                                questions=questions)
+
+    @pytest.mark.skipif(sys.version_info.minor <= 6,
+                        reason='Openai not supported for older versions')
+    def test_predict_call_dict(self):
+        params = self.get_model_params()
+        questions = self.QUESTION
+        sys_prompt = self.SYS_PROMPT
+        history = self.HISTORY
+        expected_content = self.EXPECTED_ANSWER
+        mock_result = self.create_mock_result(expected_content, params['is_openai_1_0'])
+
+        # Only prompt
```

Comment:
these cases look exactly like the ones below in test_predict_call_dict_of_str, can we refactor them out as well?